frontends/common/options.cpp

```diff
@@ -284,14 +284,16 @@ FILE* CompilerOptions::preprocess() {
 #else
         std::string cmd("cpp");
 #endif
+        const std::string arch = "-include arch.p4";
         // the p4c driver sets environment variables for include
         // paths.  check the environment and add these to the command
         // line for the preprocessor
         char * driverP4IncludePath =
           isv1() ? getenv("P4C_14_INCLUDE_PATH") : getenv("P4C_16_INCLUDE_PATH");
         cmd += cstring(" -C -undef -nostdinc") + " " + preprocessor_options
             + (driverP4IncludePath ? " -I" + cstring(driverP4IncludePath) : "")
-            + " -I" + (isv1() ? p4_14includePath : p4includePath) + " " + file;
+            + " -I" + (isv1() ? p4_14includePath : p4includePath)
+            + " " + (isv1() ? "" : arch) + " " + file;
```

Comment:
I think that this is not a good idea. A compiler should not magically include files based on esoteric and undocumented environment variables. It will be difficult for the user to understand what is going on. The compiler should include exactly the files that the user specifies in the program using `#include`. Then the compilation process is clear to everyone.

Reply:
If this is an issue about using the `p4c` driver program vs calling the backend executable directly, perhaps it should be the driver program adding the argument to its invocation of the preprocessor?  Actually, isn't it the driver that calls the preprocessor anyways and calls the backend executable with some flag to disable the preprocessor?  Or do we end up running the preprocessor twice when we use `p4c`?

Reply:
Yes, the driver would be the natural place for this.

Reply:
The driver calls the preprocessor, but the preprocessor also is called from the compiler. This is why we need deeper changes in the way we handle these includes.